src/Data/Singletons/Single.hs

```diff
@@ -666,6 +667,13 @@ singLit (IntegerL n)
                  (singFamily `DAppT` DLitT (NumTyLit n)))
   | otherwise = do sLit <- singLit (IntegerL (-n))
                    return $ DVarE sNegateName `DAppE` sLit
+singLit (StringL str) = do
+  let sing_str_lit = DVarE singMethName `DSigE`
+                     (singFamily `DAppT` DLitT (StrTyLit str))
+  os_enabled <- qIsExtEnabled LangExt.OverloadedStrings
+  pure $ if os_enabled
+         then DVarE sFromStringName `DAppE` sing_str_lit
+         else sing_str_lit
 singLit lit = do
   prom_lit <- promoteLitExp lit
```

Comment:
This will never work, I think. Remove.

Reply:
OK, I've turned this line into an error.